Tutorials/Communication_Protocols.md

```diff
@@ -157,13 +157,15 @@ The numbers in green on the diagram above delineate each bit in the byte being t
 It sounds complicated, but remember that the Tessel takes care of all of this pin manipulation for you. All you have to do is write some Javascript like this code snippet, which demonstrates the use of the SPI protocol on port A.
 
 ```js
-var portA = tessel.port['A'];
-var spi = new portA.SPI({
+const tessel = require('tessel');
+// Unpack port A from tessel.port and name it "port"
+const { A: port } = tessel.port;
+const spi = new port.SPI({
   clockSpeed: 4000000 // 4MHz
 });
 
-spi.transfer(new Buffer([0xde, 0xad, 0xbe, 0xef]), function (error, dataReceived) {
-  console.log('buffer returned by SPI slave:', dataReceived);
+spi.transfer(new Buffer([0xde, 0xad, 0xbe, 0xef]), (error, buffer) => {
+  console.log(`buffer returned by SPI slave: <${[...buffer]}>`);
```

Comment:
This is called "spread syntax", it's effectively the opposite of "rest syntax". It will take any iterable and "spread" it out into an array. Since Buffer inherits from TypedArray, it is an iterable, so it can be "spread" :)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator